### ISSUE TITLE
runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by GuangTianLi** [argoproj/argo-cd#7780]
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

argo-server pod error.

**To Reproduce**

A list of the steps required to reproduce the issue. Best of all, give us the URL to a repository that exhibits this issue.

**Expected behavior**

**Screenshots**

**Version**

```shell
argocd: v2.1.5+a8a6fc8
  BuildDate: 2021-10-20T15:16:40Z
  GitCommit: a8a6fc8dda0e26bb1e0b893e270c1128038f5b0f
  GitTreeState: clean
  GoVersion: go1.16.5
  Compiler: gc
  Platform: linux/amd64
```

**Logs**

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x574482]

goroutine 1704 [running]:
bufio.(*Writer).Flush(0xc000f204c0, 0x1, 0x7fe1af2db310)
	/usr/local/go/src/bufio/bufio.go:607 +0x62
net/http.(*response).Flush(0xc0008440e0)
	/usr/local/go/src/net/http/server.go:1681 +0x35
github.com/argoproj/pkg/grpc/http.writeKeepalive(0x3c493c0, 0xc0008440e0, 0xc00100e1d8)
	/go/pkg/mod/github.com/argoproj/pkg@v0.9.1/grpc/http/forwarders.go:151 +0x182
github.com/argoproj/pkg/grpc/http.keepalive(0x3c51690, 0xc000f16dc0, 0x3c493c0, 0xc0008440e0, 0xc00100e1d8)
	/go/pkg/mod/github.com/argoproj/pkg@v0.9.1/grpc/http/forwarders.go:166 +0x9f
created by github.com/argoproj/pkg/grpc/http.withKeepalive
	/go/pkg/mod/github.com/argoproj/pkg@v0.9.1/grpc/http/forwarders.go:174 +0x9a
```

**Comment by jomenxiao** [comment]
Fixed by argoproj/pkg#47

**Comment by alexmt** [comment]
Fix is available in v2.1.8 and v2.2.0
